nixos/modules/services/misc/autosuspend.nix

```diff
@@ -0,0 +1,291 @@
+        '';
+        example = lib.literalExpression ''
+          {
+            # Basic activity check configuration.
+            # The check class name is derived from the section header (Ping in this case).
+            # Remember to enable desired checks. They are disabled by default.
+            Ping = {
+              enabled = true;
+              hosts = "192.168.0.7";
+            };
+
+            # This check is disabled.
+            Smb.enabled = false;
+
+            # Example for a custom check name.
+            # This will use the Users check with the custom name RemoteUsers.
+            # Custom names are necessary in case a check class is used multiple times.
+            # Custom names can also be used for clarification.
+            RemoteUsers = {
+              class = "Users";
+              enabled = true;
+              name = ".*";
+              terminal = ".*";
+              host = "[0-9].*";
+            };
+
+            # Here the Users activity check is used again with different settings and a different name
+            LocalUsers = {
+              class = "Users";
+              enabled = true;
+              name = ".*";
+              terminal = ".*";
+              host = "localhost";
+            };
+          }
+        '';
+      };
+
+      wakeups = mkOption {
+        default = {};
+        type = with types; attrsOf wakeupType;
+        description = mdDoc ''
+          Checks for wake up.  For more information, see:
+          - [](https://autosuspend.readthedocs.io/en/v${version}/configuration_file.html#wake-up-check-configuration)
+          - [](https://autosuspend.readthedocs.io/en/v${version}/available_wakeups.html)
+        '';
+        example = lib.literalExpression ''
+          {
+            # Wake up checks reuse the same configuration mechanism as activity checks.
+            Calendar = {
+              enabled = true;
+              url = "http://example.org/test.ics";
+            };
+          }
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.autosuspend = {
+      description = "A daemon to suspend your server in case of inactivity";
+      documentation = ["https://autosuspend.readthedocs.io/en/latest/systemd_integration.html"];
+      wantedBy = ["multi-user.target"];
+      after = ["network.target"];
+      serviceConfig = {
+        ExecStart = ''${autosuspend}/bin/autosuspend -l ${autosuspend}/etc/autosuspend-logging.conf -c ${autosuspend-conf} daemon'';
+      };
+    };
+
+    systemd.services.autosuspend-detect-suspend = {
+      description = "Notifies autosuspend about suspension";
+      documentation = ["https://autosuspend.readthedocs.io/en/latest/systemd_integration.html"];
+      wantedBy = ["sleep.target"];
+      after = ["sleep.target"];
+      serviceConfig = {
+        ExecStart = ''${autosuspend}/bin/autosuspend -l ${autosuspend}/etc/autosuspend-logging.conf -c ${autosuspend-conf} presuspend'';
+      };
+    };
```

Comment:
I copied this directly from `autosuspend` itself, which provides config files for those two services, and just adapted it into NixOS options.  I can replace it with `powerManagement` if you think that's a better idea---this does seem a bit overkill.

In this case that would be `powerManagement.powerDownCommands` though I imagine, since it's before suspending.